src/main/java/uk/gov/hmcts/ccd/domain/service/search/CaseSearchResultGenerator.java

```diff
@@ -0,0 +1,177 @@
+            .flatMap(searchResultField -> caseTypeDefinition.getCaseFieldDefinitions().stream()
+                .filter(caseField -> caseField.getId().equals(searchResultField.getCaseFieldId()))
+                .filter(caseField -> filterDistinctFieldsByRole(addedFields, searchResultField))
+                .map(caseField -> buildSearchResultViewColumn(searchResultField, caseField))
+            )
+            .collect(Collectors.toList());
+    }
+
+    private SearchResultViewHeader buildSearchResultViewColumn(final SearchResultField searchResultField,
+                                                               final CaseFieldDefinition caseFieldDefinition) {
+        CommonField commonField = commonField(searchResultField, caseFieldDefinition);
+        return new SearchResultViewHeader(
+            searchResultField.buildCaseFieldId(),
+            commonField.getFieldTypeDefinition(),
+            searchResultField.getLabel(),
+            searchResultField.getDisplayOrder(),
+            searchResultField.isMetadata(),
+            displayContextParameter(searchResultField, commonField));
+    }
+
+    private boolean filterDistinctFieldsByRole(final HashSet<String> addedFields, final SearchResultField resultField) {
+        String id = resultField.buildCaseFieldId();
+        if (addedFields.contains(id)) {
+            return false;
+        } else {
+            if (StringUtils.isEmpty(resultField.getRole()) || userRepository.getUserRoles().contains(resultField.getRole())) {
+                addedFields.add(id);
+                return true;
+            } else {
+                return false;
+            }
+        }
+    }
+
+    private CommonField commonField(SearchResultField searchResultField, CaseFieldDefinition caseFieldDefinition) {
+        return caseFieldDefinition.getComplexFieldNestedField(searchResultField.getCaseFieldPath())
+            .orElseThrow(() ->
+                new BadRequestException(format("CaseField %s has no nested elements with code %s.",
+                    caseFieldDefinition.getId(), searchResultField.getCaseFieldPath())));
+    }
+
+    private String displayContextParameter(SearchResultField searchResultField, CommonField commonField) {
+        return searchResultField.getDisplayContextParameter() == null
+            ? commonField.getDisplayContextParameter()
+            : searchResultField.getDisplayContextParameter();
+    }
+
+    private SearchResultViewItem buildSearchResultViewItem(final CaseDetails caseDetails,
+                                                           final CaseTypeDefinition caseTypeDefinition,
+                                                           final SearchResult searchResult) {
+
+        Map<String, JsonNode> caseData = new HashMap<>(caseDetails.getData());
+        Map<String, Object> caseMetadata = new HashMap<>(caseDetails.getMetadata());
+        Map<String, TextNode> labels = caseTypeDefinition.getLabelsFromCaseFields();
```

Comment:
not sure I understand why we are getting the fields of type label for here. Furthermore, shall we rename the method to 
`getLabelFieldTypesFromCaseFields()` or sim? unless I'm misunderstanding what this is doing

Reply:
Inherited behaviour from `MergeDataToSearchResultOperation`. I'm a bit hesitant to remove in case of breaking anything the UI is depending on, so will leave unless we see at a later date that definitely not required.